pilot/pkg/model/sidecar.go

```diff
@@ -552,7 +552,14 @@ func (sc *SidecarScope) DependsOnConfig(config ConfigKey) bool {
 
 // AddConfigDependencies add extra config dependencies to this scope. This action should be done before the
 // SidecarScope being used to avoid concurrent read/write.
-func (sc *SidecarScope) AddConfigDependencies(dependencies ...ConfigHash) {
+// Used only in tests.
+func (sc *SidecarScope) AddConfigDependenciesForTesting(dependencies ...ConfigHash) {
+	sc.addConfigDependencies(dependencies...)
+}
+
+// addConfigDependencies add extra config dependencies to this scope. This action should be done before the
+// SidecarScope being used to avoid concurrent read/write.
+func (sc *SidecarScope) addConfigDependencies(dependencies ...ConfigHash) {
```

Comment:
I doubt the necessity of doing this, if we are changing this, I think we have many other methods/function that can be rewrite.

Reply:
In general I think if it is just used in tests, it is better to follow this pattern so that we do not unnecessarily expose these. Wherever I am seeing this, I am changing if I am changing code near this. But I am not so strong about this - If you think, it is not good, I can revert this.